client/src/components/Search/Filters/FilterList.js

```diff
@@ -0,0 +1,184 @@
+                                        decades={decades}
+                                        onChange={(params) => this.handleDateSearchSelect(params, index)}>
+                                    </DateFilter>))
+                        }
+                        {
+                            this.state.filterList.location.map((params, index) => (
+                                <div className="filter-body" key={index.toString()}>
+                                    <button style={{backgroundColor: "white", border: "none"}} onClick={() => this.removeFilter()}>
+                                        <i className="material-icons ">clear</i>
+                                    </button>
+                                    <LocationSearch
+                                        value={params}
+                                        onChange={(params) => this.handleLocationSearchSelect(params)}>
+                                    </LocationSearch>
+                                </div>))
+                        }
+                        {
+                            this.state.filterList.materials.map((params, index) => (
+                                <div className="filter-body" key={index.toString()}>
+                                    <button style={{backgroundColor: "white", border: "none"}} onClick={() => this.removeFilter()}>
+                                        <i className="material-icons ">clear</i>
+                                    </button>
+                                    <TagsSearch
+                                        variant="materials"
+                                        tags={params}
+                                        allowTagCreation={false}
+                                        onChange={(variant, params) => this.handleTagsSearchTagSelect(variant, params)}>
+                                    </TagsSearch>
+                                </div>))
+                        }
+                        {
+                            this.state.filterList.tags.map((params, index) => (
+                                <div className="filter-body" key={index.toString()}>
+                                    <button style={{backgroundColor: "white", border: "none"}} onClick={() => this.removeFilter()}>
+                                        <i className="material-icons ">clear</i>
+                                    </button>
+                                    <TagsSearch
+                                        variant="tags"
+                                        onRemove= {() => this.removeFilter('tags', index)}
+                                        tags={params}
+                                        allowTagCreation={false}
+                                        onChange={(variant, params) => this.handleTagsSearchTagSelect(variant, params)}>
+                                    </TagsSearch>
+                                </div>))
+                        }
```

Comment:
maybe break these map functions into their own functions and call them from here